ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js

```diff
@@ -270,6 +290,158 @@ module.exports = Backbone.AssociatedModel.extend({
     }
   },
 
+  convertLatLonLinePolyToUsng(points) {
+    return points
+      ? points.map((point) => {
+          // A little bit unintuitive, but lat/lon is swapped here
+          return converter.LLtoMGRSUPS(point[1], point[0], usngPrecision)
+        })
+      : undefined
+  },
+
+  convertLatLonLinePolyToDms(points) {
+    return points
+      ? points.map((point) => {
+          const lat = dmsUtils.ddToDmsCoordinateLat(point[1])
+          const lon = dmsUtils.ddToDmsCoordinateLon(point[0])
+          return {
+            lat: lat.coordinate,
+            lon: lon.coordinate,
+            latDirection: lat.direction,
+            lonDirection: lon.direction,
+          }
+        })
+      : undefined
+  },
+
+  convertLatLonLinePolyToUtm(points) {
+    return points
+      ? points.map((point) => {
+          let llPoint = this.LLtoUtmUps(point[1], point[0])
+          return {
+            ...llPoint,
+            hemisphere:
+              llPoint.hemisphere.toUpperCase() === 'NORTHERN'
+                ? 'Northern'
+                : 'Southern',
+          }
+        })
+      : undefined
+  },
+
+  setUsngDmsUtmWithLineOrPoly(model) {
+    let key = this.get('mode')
+    if (key === 'poly') key = 'polygon'
+    if (key && (key === 'line' || key === 'polygon')) {
+      const points = this.get(key)
+      const usngPoints = this.convertLatLonLinePolyToUsng(points)
+      const dmsPoints = this.convertLatLonLinePolyToDms(points)
+      const utmupsPoints = this.convertLatLonLinePolyToUtm(points)
+      model.set(
+        {
+          usngPointArray: usngPoints,
+          dmsPointArray: dmsPoints,
+          utmUpsPointArray: utmupsPoints,
+        },
+        {
+          silent: true, //don't trigger another onChange for line or poly
```

Comment:
📝  See https://backbonejs.org/#Events-catalog for details on {silent: true}
> Generally speaking, when calling a function that emits an event (model.set, collection.add, and so on...), if you'd like to prevent the event from being triggered, you may pass {silent: true} as an option. Note that this is rarely, perhaps even never, a good idea. Passing through a specific flag in the options for your event callback to look at, and choose to ignore, will usually work out better.

The reason we need to use it here is because we are listening to the same thing we are changing.
The input for location is listening to changes from the map, but it is also changing the map. This creates a weird loop- so we want to make sure we stop the onChange trigger here.